dashboard/app/controllers/print_certificates_controller.rb

```diff
@@ -20,4 +20,19 @@ def show
     @student_name = data['name']
     @certificate_image_url = certificate_image_url(data['name'], data['course'], data['donor'])
   end
+
+  # POST /print_certificates/batch
+  def batch
+    view_options(no_header: true, no_footer: true, white_background: true, full_width: true)
+
+    student_names = params[:studentNames]&.split("\n")&.map(&:strip)&.shift(30)
```

Comment:
I see the instruction above that this will only print 30 names, but it looks like this will still "work" if there are more than 30 names, with no warning that some names didn't get printed. Is there a plan for a warning/error?

Reply:
I'm also curious where this limit comes from –– teachers can easily have 120 students

Reply:
good points. to answer questions literally:

1. there is this "warning" ahead of time:
![Screen Shot 2022-09-13 at 12 50 49 PM](https://user-images.githubusercontent.com/8001765/189996123-c81e537c-2406-4ab9-abb0-2213c667fefd.png)

2. there is not a plan to add a better warning here. the specced behavior is just to match parity with the existing pages.

I'm not sure where the earlier limit came from, but I think we do want some upper bound, since generating images can be expensive for our servers, and attacks are often measured by how much load you can put on a server per web request.

@tess323 do you have a thought here on whether a better warning experience is worth doing?

Reply:
Got it, sounds good!